autogen/oai/client.py

```diff
@@ -144,10 +349,26 @@ def _client(self, config: Dict[str, Any], openai_config: Dict[str, Any]) -> Open
             if openai_config["azure_deployment"] is not None:
                 openai_config["azure_deployment"] = openai_config["azure_deployment"].replace(".", "")
             openai_config["azure_endpoint"] = openai_config.get("azure_endpoint", openai_config.pop("base_url", None))
-            client = AzureOpenAI(**openai_config)
-        else:
-            client = OpenAI(**openai_config)
-        return client
+            self._clients.append(OpenAIClient(AzureOpenAI(**openai_config)))
+        elif api_type is None:
+            self._clients.append(OpenAIClient(OpenAI(**openai_config)))
+        # else a config for a custom client is set
+        # skipping until the register_custom_client is called with the appropriate class
+
+    def register_custom_client(self, ClientClass: Client, **kwargs):
+        """Register a custom client.
+
+        Args:
+            client: A custom client that follows the Client interface
+        """
+        found = False
+        for config in self._config_list:
+            if config["api_type"] is not None and config["api_type"] == ClientClass.__name__:
+                client = ClientClass(config, **kwargs)
+                self._clients.append(client)
+                found = True
+        if not found:
+            raise ValueError(f'Custom client "{ClientClass.__name__}" was not found in the config_list.')
```

Comment:
To help caller understand what to do to fix this issue, we can add something like "Please make sure to include an entry in the config_list with api_type = ClientClass.__name__.

Reply:
added